### ISSUE TITLE
Nested loop causing app to crash

### ISSUE THREAD

**Issue by wileybaba** [lukehaas/runjs#41]
I was curious how the RunJS app would respond to a nested infinite loop and caused the application to stop responding with the following code 
```
for (i=0; i<1000; i++) {
  for(j=0; j<1000; j++) {
    console.log('test')
  }
}
```

I had to rm rf RunJS and remove its Application Support file, then reinstall in order to get the app running properly again.  I was able to reproduce the bug easily.  Would someone mind trying to reproduce this on their machine?

**Comment by townofdon** [comment]
Reproduced on mine, accidentally typed the following code that got evaluated instantly before I could prevent infinite loop (note the lack of curly braces):

```
for (let i = 0; i < 1000000; i++)
benchmark();
```

So it's not truly infinite but I had to wait several minutes for it to finish evaluating. It would be really useful if there was a way that a script could be halted or prevented from running when the app loads.

Would be even better if infinite loops were somehow detected.

Problems:

- entire app freezes on infinite loop, so you can't edit preferences to turn off auto-evaluation
- same code shows up in editor and is re-evaluated upon reopening the app

**Comment by townofdon** [comment]
This is an awesome little app by the way!

**Comment by lukehaas** [comment]
Thanks. This is a high priority issue. I'm aiming to have a solution soon.

**Comment by jkodu** [comment]
Ran into the same problem @lukehaas. Nice little tool btw.

**Comment by lukehaas** [comment]
This issue is now resolved in version 1.2.3

![screenshot 2019-02-02 at 17 39 27](https://user-images.githubusercontent.com/741344/52167446-f8e28480-2712-11e9-95a7-633f1ad6b079.png)
